### ISSUE TITLE
[Bug] InvalidInjectionException: Invalid descriptor on betterend.mixins.common.json after updated to betterend 0.9.1

### ISSUE THREAD

**Issue by KiteJencien** [paulevsGitch/BetterEnd#102]
## Environment Description:
- java version: jdk-14
- Minecraft version: 1.16.4 release , IntellijIDEA fabric development environment
- Betterend mod version: betterend 0.9.1: 1.16.4/5

## Log for Exception:
```
[20:56:00] [main/WARN]: @Mixin target net/minecraft/village/TradeOffers$TypeAwareBuyForOneEmeraldFactory is public in fabric-object-builder-v1.mixins.json:TypeAwareTradeMixin and should be specified in value
[20:56:00] [main/WARN]: @Mixin target net/minecraft/client/font/FontManager$1 is public in fabric-resource-loader-v0.mixins.json:client.FontManagerResourceReloadListenerMixin and should be specified in value
[20:56:00] [main/WARN]: @Mixin target net/minecraft/server/command/GameRuleCommand$1 is public in fabric-game-rule-api-v1.mixins.json:GameRuleCommandVisitorMixin and should be specified in value
[20:56:00] [main/WARN]: @Mixin target net/minecraft/client/gui/screen/world/EditGameRulesScreen$RuleListWidget$1 is public in fabric-game-rule-api-v1.mixins.json:client.RuleListWidgetVisitorMixin and should be specified in value
[20:56:00] [main/WARN]: @Mixin target net/minecraft/server/world/ThreadedAnvilChunkStorage$EntityTracker is public in fabric-networking-api-v1.mixins.json:accessor.EntityTrackerAccessor and should be specified in value
[20:56:00] [main/WARN]: @Mixin target net/minecraft/client/particle/ParticleManager$SimpleSpriteProvider is public in fabric-particles-v1.mixins.json:ParticleManagerAccessor$SimpleSpriteProviderAccessor and should be specified in value
[20:56:00] [main/WARN]: @Mixin target net/minecraft/tag/Tag$1 is public in fabric-tag-extensions-v0.mixins.json:MixinTagImpl and should be specified in value
[20:56:00] [main/WARN]: @Mixin target net/minecraft/tag/RequiredTagList$TagWrapper is public in fabric-tag-extensions-v0.mixins.json:MixinTagImpl and should be specified in value
[20:56:03] [main/FATAL]: Mixin apply failed betterend.mixins.common.json:AnvilScreenHandlerMixin -> net.minecraft.screen.AnvilScreenHandler: org.spongepowered.asm.mixin.injection.throwables.InvalidInjectionException Invalid descriptor on betterend.mixins.common.json:AnvilScreenHandlerMixin->@Inject::be_initAnvilLevel(ILnet/minecraft/entity/player/PlayerInventory;Lnet/minecraft/screen/ScreenHandlerContext;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V! Expected (ILnet/minecraft/entity/player/PlayerInventory;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V but found (ILnet/minecraft/entity/player/PlayerInventory;Lnet/minecraft/screen/ScreenHandlerContext;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V [INJECT Applicator Phase -> betterend.mixins.common.json:AnvilScreenHandlerMixin -> Apply Injections ->  -> Inject -> betterend.mixins.common.json:AnvilScreenHandlerMixin->@Inject::be_initAnvilLevel(ILnet/minecraft/entity/player/PlayerInventory;Lnet/minecraft/screen/ScreenHandlerContext;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V]
org.spongepowered.asm.mixin.injection.throwables.InvalidInjectionException: Invalid descriptor on betterend.mixins.common.json:AnvilScreenHandlerMixin->@Inject::be_initAnvilLevel(ILnet/minecraft/entity/player/PlayerInventory;Lnet/minecraft/screen/ScreenHandlerContext;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V! Expected (ILnet/minecraft/entity/player/PlayerInventory;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V but found (ILnet/minecraft/entity/player/PlayerInventory;Lnet/minecraft/screen/ScreenHandlerContext;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V [INJECT Applicator Phase -> betterend.mixins.common.json:AnvilScreenHandlerMixin -> Apply Injections ->  -> Inject -> betterend.mixins.common.json:AnvilScreenHandlerMixin->@Inject::be_initAnvilLevel(ILnet/minecraft/entity/player/PlayerInventory;Lnet/minecraft/screen/ScreenHandlerContext;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V]
	at org.spongepowered.asm.mixin.injection.callback.CallbackInjector.inject(CallbackInjector.java:517) ~[sponge-mixin-0.9.2+mixin.0.8.2.jar:0.9.2+mixin.0.8.2]
	at org.spongepowered.asm.mixin.injection.callback.CallbackInjector.inject(CallbackInjector.java:447) ~[sponge-mixin-0.9.2+mixin.0.8.2.jar:0.9.2+mixin.0.8.2]
	at org.spongepowered.asm.mixin.injection.code.Injector.inject(Injector.java:264) ~[sponge-mixin-0.9.2+mixin.0.8.2.jar:0.9.2+mixin.0.8.2]
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.inject(InjectionInfo.java:385) ~[sponge-mixin-0.9.2+mixin.0.8.2.jar:0.9.2+mixin.0.8.2]
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1284) ~[sponge-mixin-0.9.2+mixin.0.8.2.jar:0.9.2+mixin.0.8.2]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1042) ~[sponge-mixin-0.9.2+mixin.0.8.2.jar:0.9.2+mixin.0.8.2]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:395) ~[sponge-mixin-0.9.2+mixin.0.8.2.jar:0.9.2+mixin.0.8.2]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:320) ~[sponge-mixin-0.9.2+mixin.0.8.2.jar:0.9.2+mixin.0.8.2]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:345) ~[sponge-mixin-0.9.2+mixin.0.8.2.jar:0.9.2+mixin.0.8.2]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:569) ~[sponge-mixin-0.9.2+mixin.0.8.2.jar:0.9.2+mixin.0.8.2]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:351) ~[sponge-mixin-0.9.2+mixin.0.8.2.jar:0.9.2+mixin.0.8.2]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:208) ~[sponge-mixin-0.9.2+mixin.0.8.2.jar:0.9.2+mixin.0.8.2]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:178) ~[sponge-mixin-0.9.2+mixin.0.8.2.jar:0.9.2+mixin.0.8.2]
	at org.spongepowered.asm.mixin.transformer.FabricMixinTransformerProxy.transformClassBytes(FabricMixinTransformerProxy.java:23) ~[fabric-loader-0.11.2@net.fabricmc.yarn.1_16_4.1.16.4+build.9-v2.jar:0.9.2+mixin.0.8.2]
	at net.fabricmc.loader.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:157) ~[fabric-loader-0.11.2@net.fabricmc.yarn.1_16_4.1.16.4+build.9-v2.jar:?]
	at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:150) ~[fabric-loader-0.11.2@net.fabricmc.yarn.1_16_4.1.16.4+build.9-v2.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:521) ~[?:?]
	at net.minecraft.block.Blocks.<clinit>(Blocks.java:426) ~[minecraft-1.16.4-mapped-net.fabricmc.yarn-1.16.4+build.9-v2.jar:?]
	at net.minecraft.world.gen.surfacebuilder.SurfaceBuilder.<clinit>(SurfaceBuilder.java:13) ~[minecraft-1.16.4-mapped-net.fabricmc.yarn-1.16.4+build.9-v2.jar:?]
	at net.minecraft.world.gen.surfacebuilder.ConfiguredSurfaceBuilders.<clinit>(ConfiguredSurfaceBuilders.java:11) ~[minecraft-1.16.4-mapped-net.fabricmc.yarn-1.16.4+build.9-v2.jar:?]
	at net.minecraft.util.registry.BuiltinRegistries.method_30573(BuiltinRegistries.java:44) ~[minecraft-1.16.4-mapped-net.fabricmc.yarn-1.16.4+build.9-v2.jar:?]
	at net.minecraft.util.registry.BuiltinRegistries.method_30566(BuiltinRegistries.java:75) ~[minecraft-1.16.4-mapped-net.fabricmc.yarn-1.16.4+build.9-v2.jar:?]
	at java.util.LinkedHashMap.forEach(LinkedHashMap.java:684) [?:?]
	at net.minecraft.util.registry.BuiltinRegistries.<clinit>(BuiltinRegistries.java:74) [minecraft-1.16.4-mapped-net.fabricmc.yarn-1.16.4+build.9-v2.jar:?]
	at net.minecraft.util.registry.Registry.<clinit>(Registry.java:237) [minecraft-1.16.4-mapped-net.fabricmc.yarn-1.16.4+build.9-v2.jar:?]
	at net.minecraft.Bootstrap.initialize(Bootstrap.java:42) [minecraft-1.16.4-mapped-net.fabricmc.yarn-1.16.4+build.9-v2.jar:?]
Exception in thread "main" 	at net.minecraft.client.main.Main.main(Main.java:137) [minecraft-1.16.4-mapped-net.fabricmc.yarn-1.16.4+build.9-v2.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:226) [fabric-loader-0.11.2@net.fabricmc.yarn.1_16_4.1.16.4+build.9-v2.jar:?]
	at net.fabricmc.loader.launch.knot.Knot.launch(Knot.java:146) [fabric-loader-0.11.2@net.fabricmc.yarn.1_16_4.1.16.4+build.9-v2.jar:?]
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:28) [fabric-loader-0.11.2@net.fabricmc.yarn.1_16_4.1.16.4+build.9-v2.jar:?]
	at net.fabric.example.LaunchInjection.main(LaunchInjection.java:67) [main/:?]
	at net.fabric.example.Launcher.main(Launcher.java:43) [main/:?]
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:228)
	at net.fabricmc.loader.launch.knot.Knot.launch(Knot.java:146)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:28)
	at net.fabric.example.LaunchInjection.main(LaunchInjection.java:67)
	at net.fabric.example.Launcher.main(Launcher.java:43)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
Caused by: java.lang.reflect.InvocationTargetException

	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:226)
	... 4 more
Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:208)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:178)
	at org.spongepowered.asm.mixin.transformer.FabricMixinTransformerProxy.transformClassBytes(FabricMixinTransformerProxy.java:23)
	at net.fabricmc.loader.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:157)
	at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:150)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	at net.minecraft.block.Blocks.<clinit>(Blocks.java:426)
	at net.minecraft.world.gen.surfacebuilder.SurfaceBuilder.<clinit>(SurfaceBuilder.java:13)
	at net.minecraft.world.gen.surfacebuilder.ConfiguredSurfaceBuilders.<clinit>(ConfiguredSurfaceBuilders.java:11)
	at net.minecraft.util.registry.BuiltinRegistries.method_30573(BuiltinRegistries.java:44)
	at net.minecraft.util.registry.BuiltinRegistries.method_30566(BuiltinRegistries.java:75)
	at java.base/java.util.LinkedHashMap.forEach(LinkedHashMap.java:684)
	at net.minecraft.util.registry.BuiltinRegistries.<clinit>(BuiltinRegistries.java:74)
	at net.minecraft.util.registry.Registry.<clinit>(Registry.java:237)
	at net.minecraft.Bootstrap.initialize(Bootstrap.java:42)
	at net.minecraft.client.main.Main.main(Main.java:137)
	... 9 more
Caused by: org.spongepowered.asm.mixin.throwables.MixinApplyError: Mixin [betterend.mixins.common.json:AnvilScreenHandlerMixin] from phase [DEFAULT] in config [betterend.mixins.common.json] FAILED during APPLY
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.handleMixinError(MixinProcessor.java:642)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.handleMixinApplyError(MixinProcessor.java:594)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:356)
	... 25 more
Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered

Caused by: org.spongepowered.asm.mixin.injection.throwables.InvalidInjectionException: Invalid descriptor on betterend.mixins.common.json:AnvilScreenHandlerMixin->@Inject::be_initAnvilLevel(ILnet/minecraft/entity/player/PlayerInventory;Lnet/minecraft/screen/ScreenHandlerContext;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V! Expected (ILnet/minecraft/entity/player/PlayerInventory;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V but found (ILnet/minecraft/entity/player/PlayerInventory;Lnet/minecraft/screen/ScreenHandlerContext;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V [INJECT Applicator Phase -> betterend.mixins.common.json:AnvilScreenHandlerMixin -> Apply Injections ->  -> Inject -> betterend.mixins.common.json:AnvilScreenHandlerMixin->@Inject::be_initAnvilLevel(ILnet/minecraft/entity/player/PlayerInventory;Lnet/minecraft/screen/ScreenHandlerContext;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V]
	at org.spongepowered.asm.mixin.injection.callback.CallbackInjector.inject(CallbackInjector.java:517)
	at org.spongepowered.asm.mixin.injection.callback.CallbackInjector.inject(CallbackInjector.java:447)
Caused by: org.spongepowered.asm.mixin.throwables.MixinApplyError: Mixin [betterend.mixins.common.json:AnvilScreenHandlerMixin] from phase [DEFAULT] in config [betterend.mixins.common.json] FAILED during APPLY

	at org.spongepowered.asm.mixin.injection.code.Injector.inject(Injector.java:264)
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.inject(InjectionInfo.java:385)
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1284)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1042)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:395)
Caused by: org.spongepowered.asm.mixin.injection.throwables.InvalidInjectionException: Invalid descriptor on betterend.mixins.common.json:AnvilScreenHandlerMixin->@Inject::be_initAnvilLevel(ILnet/minecraft/entity/player/PlayerInventory;Lnet/minecraft/screen/ScreenHandlerContext;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V! Expected (ILnet/minecraft/entity/player/PlayerInventory;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V but found (ILnet/minecraft/entity/player/PlayerInventory;Lnet/minecraft/screen/ScreenHandlerContext;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V [INJECT Applicator Phase -> betterend.mixins.common.json:AnvilScreenHandlerMixin -> Apply Injections ->  -> Inject -> betterend.mixins.common.json:AnvilScreenHandlerMixin->@Inject::be_initAnvilLevel(ILnet/minecraft/entity/player/PlayerInventory;Lnet/minecraft/screen/ScreenHandlerContext;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V]

	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:320)
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:345)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:569)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:351)
	... 25 more

> Task :Launcher.main() FAILED

Execution failed for task ':Launcher.main()'.
> Process 'command 'C:/Program Files/Java/jdk-11.0.8/bin/java.exe'' finished with non-zero exit value 1
```

## Dependencies and Configurations:
Gradle.properties:
```
# Done to increase the memory available to gradle.
org.gradle.jvmargs=-Xmx1G

# Fabric Properties
# check these on https://fabricmc.net/use
minecraft_version=1.16.4
yarn_mappings=1.16.4+build.9
loader_version=0.11.2


#Fabric api

# Mod Properties
mod_version = 1.0.0
maven_group = io.kitejencien
archives_base_name = fabric-example-mod

# Dependencies
# currently not on the main fabric site, check on the maven: https://maven.fabricmc.net/net/fabricmc/fabric-api/fabric-api
fabric_version=0.31.0+1.16
```

Dependencies (latest) :
```gradle
dependencies {
	minecraft "com.mojang:minecraft:1.16.4"
	mappings "net.fabricmc:yarn:1.16.4+build.9:v2"
	modImplementation "net.fabricmc:fabric-loader:0.11.2"

	//Fabric api
	modImplementation "net.fabricmc.fabric-api:fabric-api:0.31.0+1.16"
}
```

Mixin options:
```json
{
  "required": true,
  "minVersion": "0.8.2",
  "package": "net.fabric.example.mixin",
  "compatibilityLevel": "JAVA_8",
  "mixins": [
  ],
  "client": [
    "ExampleMixin"
  ],
  "injectors": {
    "defaultRequire": 1
  }
}
```

## Additional Info:
I installed the mod as normal players inside the mods folder ( since the server I'm playing and testing on requires this mod), the beta version jars would launch just fine but the 0.9.1 version will simply crash the game during initialization.

I've updated all the 1.16.4 dependencies in the project files to the latest version.

// The `Launcher.java` and `LaunchInjection.java` are just files I used to parse some authentication arguments into the game, just ignore them, and the game works the same way as normal `:runclient` task

**Comment by Bulldog83** [comment]
In dev code

**Comment by paulevsGitch** [comment]
Fixed in 0.9.2

**Comment by KiteJencien** [comment]
THX
